src/Symfony/Component/HttpFoundation/JsonResponse.php

```diff
@@ -171,6 +171,11 @@ public function setEncodingOptions($encodingOptions)
     {
         $this->encodingOptions = (int) $encodingOptions;
 
+        // If content is set, update $this->data with decoded content
+        if (!empty($this->content)) {
+            return $this->setData(json_decode($this->content));
```

Comment:
this is broken when the JSONP callback is used, as the content is *not* a JSON string in this case